src/OpenTracing/Propagation/TextMapExtractAdapter.cs

```diff
@@ -0,0 +1,55 @@
+﻿// --------------------------------------------------------------------------------------------------------------------
+// <copyright file="TextMapExtractAdapter.cs">
+//   Copyright 2017-2018 The OpenTracing Authors
+//   
+//   Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+//   in compliance with the License. You may obtain a copy of the License at
+//   
+//   http://www.apache.org/licenses/LICENSE-2.0
+// 
+//   Unless required by applicable law or agreed to in writing, software distributed under the License
+//   is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+//   or implied. See the License for the specific language governing permissions and limitations under
+//   the License.
+// </copyright>
+// --------------------------------------------------------------------------------------------------------------------
+
+namespace OpenTracing.Propagation
+{
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+
+    /// <summary>
+    ///     A TextMap carrier for use with Tracer.extract() ONLY (it has no mutating methods).
+    ///     Note that the TextMap interface can be made to wrap around arbitrary data types (not just Dictionary{string,
+    ///     string}
+    ///     as illustrated here).
+    /// </summary>
+    /// <seealso cref="ITracer.Extract{T}" />
+    public sealed class TextMapExtractAdapter : TextMap
+    {
+        private readonly IDictionary<string, string> dictionary;
```

Comment:
`_dictionary`. See https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/coding-style.md for more details on what I'd like to stick to. We should link this in a contributing-file.

Reply:
resolved